### PR TITLE
HPCC-26947 Add --fetchrepos and --updaterepos to ecl command

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
HPCC-26493 Has a note to include ecl command in changes to the documentation

File: ecl/eclcc/eclcc.hpp [hunk 1]
```diff
@@ -85,12 +85,12 @@ const char * const helpText[] = {
     "!   -brk <n>      Trigger a break point in eclcc after nth allocation",
 #endif
     "!   -Dname=value  Override the definition of a global attribute 'name'",
-    "!   --defaultgitprefix <prefix>  The default prefix used to access git repos when not specified in package.json",
+    "!?  --defaultgitprefix <prefix>  The default prefix used to access git repos when not specified in package.json",
```
Comment (afishbeck) [inline ecl/eclcc/eclcc.hpp:88]: Should the order has to be "?!" for this to be included properly in ecl command usage (and for the markup character to be removed)?

File: system/jlib/jargv.cpp [hunk 2]
```diff
@@ -218,3 +218,41 @@ bool ArgvIterator::matchPathFlag(StringBuffer & option, const char * name)
...
+        // -Ix or -I x
+        if (isShortOption)
+        {
+            option.append(arg);
+            return true;
+        }
+        return false;
+    }
+
+    option.append(arg);
```
Comment (afishbeck) [inline system/jlib/jargv.cpp:247]: Just a note.  Would matter for some string related functions, but not really here.  A return value of false does not necessarily mean that the StringBuffer was unchanged, and option.isEmpty() is not the same as a return value of false.

**Review by afishbeck — CHANGES_REQUESTED** [review]
One possible minor issue with usage.  Other than that looks good.

File: ecl/eclcc/eclcc.hpp (hunk 1 above)
Comment (ghalliday) [reply ecl/eclcc/eclcc.hpp]: well spotted.

File: system/jlib/jargv.cpp (hunk 2 above)
Comment (ghalliday) [reply system/jlib/jargv.cpp]: I have changed the code - and also picked up a couple of bugs while testing.

**Review by afishbeck — APPROVED** [review]
@ghalliday looks good.

**Pull request merged**